docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+
+If a clean sync was not requested by either the client or the sync service, then the sync service should only send new rules seen since the last time the client synced.
+
+#### `ruledownload` Request
+
+Rules are retrieved from the sync server by having the client (Santa) issues an
+HTTP POST request to the url `/ruledownload/<machine_id>`
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | a field used by the Sync Server to indicate where the next batch of rules should start |
+
+
+##### `ruledownload` Request Example Payload 
+
+On the first request the payload is an empty dictionary
+
+```json
+{}
+```
+
+The server will then send a response with the number of rules specified in the preflight response `rule_batch` the response will also contain a special field called `cursor`. The value and form of this field is left to the sync server implementor but is used to indicate that the client is ready for the next batch of rules.
+
+On subsequent requests to the server the cursor field is sent with the value from the previous response.
+
+```json
+{"cursor":"CpgBChcKCnVwZGF0ZWRfZHQSCQjh94a58uLlAhJ5ahVzfmdvb2dsZS5jb206YXBwbm90aHJyYAsSCUJsb2NrYWJsZSJAMTczOThkYWQzZDAxZGRmYzllMmEwYjBiMWQxYzQyMjY1OWM2ZjA3YmU1MmY3ZjQ1OTVmNDNlZjRhZWI5MGI4YQwLEgRSdWxlGICA8MvA0tIJDBgAIAA="}
+```
+
+#### `ruledownload` Response
+
+When a `ruledownload` request is received the sync server responds with a JSON object
+containing a list of rule objects and a cursor so the client can resume
+downloading if the rules need to be downloaded in multiple batches. 
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | used to continue a rule download in a future request |
+| rules | YES | list of Rule objects | List of rule objects |
+
+##### Rules Objects
+
+
+:warning: Rule objects must have one of either the `identifier` or `sha256` fields.  The `sha256` field is supported for backwards compatability. 
+
+| Key | Required | Type | Meaning | Example Value | 
+|---|---|---|---|---|
+| identifier | YES | string | The attribute of the binary the rule should match on e.g. the team ID of a binary or sha256 hash value | "ff2a7daa4c25cbd5b057e4471c6a22aba7d154dadfb5cce139c37cf795f41c9c" |
+| sha256 | YES if `identifier` is not used | string | The sha256 of an attribute of the binary the rule should match on sha256 hash value of the binary or one of its signing certificates | "ff2a7daa4c25cbd5b057e4471c6a22aba7d154dadfb5cce139c37cf795f41c9c" 
+| policy | YES | string | identifies the action to perform in response to the rule matching must be one of the examples. | "ALLOWLIST","ALLOWLIST_COMPILER", "BLACKLIST", "BLOCKLIST", "REMOVE", "SILENT_BLACKLIST", "SILENT_BLOCKLIST", "WHITELIST", "WHITELIST_COMPILER"  |
```

Comment:
Remove the deprecated terms from the list (BLACKLIST, WHITELIST), those remain only for backward compatibility

Reply:
OK, can do. I left them in there because it's what Santa supports today.

I'd left them in there because moroz is using them today: 

https://github.com/groob/moroz/blob/cf740df50fa91bd5b5f2a8946571fad745eafece/configs/global.toml#L10

Reply:
Yeah, I'm not sure what to do there, I'd prefer to keep them out of the documentation to avoid server authors using them in new code and to make it clear the old keys are only for backward compatibility.

Reply:
I filed a PR against moroz to clean it up https://github.com/groob/moroz/pull/30

Reply:
Will need to file another PR to get them using `identifier` instead of `SHA256`.